rust/connlib/tunnel/src/peer.rs

```diff
@@ -433,6 +447,55 @@ impl ClientOnGateway {
     pub fn id(&self) -> ClientId {
         self.id
     }
+
+    fn dst_unreachable(&self, packet: &IpPacket) -> Result<TranslateOutboundResult> {
+        let dst = packet.destination();
+
+        // TODO: Should we use the source IP of the packet?
+        let icmp_error = match dst {
+            IpAddr::V4(inside_dst) => icmpv4_unreachable(inside_dst, self.client_tun.v4, packet)?,
+            IpAddr::V6(inside_dst) => icmpv6_unreachable(inside_dst, self.client_tun.v6, packet)?,
+        };
+
+        Ok(TranslateOutboundResult::DestinationUnreachable(icmp_error))
```

Comment:
Hm, yeah I would think you want the source address here to be the Gateway's tunnel IP address right? That's the sender after all.

Reply:
Yep works but doesn't make a difference in my testing.